tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -264,11 +264,13 @@ public void Dispose()
         private void FeaturesProductConfigChanged(object sender, ProductConfigChangedEventArgs e)
         {
             var features = e.GetDeserializedConfigurations<Features>().FirstOrDefault();
-            if (features != null)
+            if (features.TypedFile != null)
             {
-                _settings.Enabled = features.Asm.Enabled;
+                _settings.Enabled = features.TypedFile.Asm.Enabled;
                 UpdateStatus();
             }
+
+            e.Acknowldge(features.Name);
```

Comment:
Should other products using RCM (i.e. Live Debugger) be calling `Acknowledge` too? Afacit we don't call `Error` anywhere? Presumably somewhere should be?

Reply:
Live debug would ideally call Acknowledge too, but it's less important for them, since they their own dedicated communication channel to send similar information, that option isn't available to ASM.

Error is only called in the error handle where the event is called, this seems to be enough, as there's an integration test dedicated to the error case that works well.